- add option to disable the port lights showing the current voltage to save cpu cycles
- add 16 and 32 khz sample rates for low end systems
- change some default settings for low end systems
  - default sample rate = 22.05 khz
  - 2 threads
  - frame swap interval = 15

diff --git a/include/settings.hpp b/include/settings.hpp
index 2b7c1a9b..7e6a1b21 100644
--- a/include/settings.hpp
+++ b/include/settings.hpp
@@ -47,6 +47,7 @@ extern float cableOpacity;
 extern float cableTension;
 extern float rackBrightness;
 extern float haloBrightness;
+extern bool disablePortLights;
 /** Allows rack to hide and lock the cursor position when dragging knobs etc. */
 extern bool allowCursorLock;
 enum KnobMode {
diff --git a/src/app/CableWidget.cpp b/src/app/CableWidget.cpp
index 4cb9dc69..edb14c26 100644
--- a/src/app/CableWidget.cpp
+++ b/src/app/CableWidget.cpp
@@ -86,18 +86,20 @@ PlugWidget::~PlugWidget() {
 }
 
 void PlugWidget::step() {
-	std::vector<float> values(3);
-
-	PortWidget* pw = internal->cableWidget->getPort(internal->type);
-	if (pw && internal->plugLight->isVisible()) {
-		engine::Port* port = pw->getPort();
-		if (port) {
-			for (int i = 0; i < 3; i++) {
-				values[i] = port->plugLights[i].getBrightness();
+	if(!settings::disablePortLights) {
+		std::vector<float> values(3);
+
+		PortWidget* pw = internal->cableWidget->getPort(internal->type);
+		if (pw && internal->plugLight->isVisible()) {
+			engine::Port* port = pw->getPort();
+			if (port) {
+				for (int i = 0; i < 3; i++) {
+					values[i] = port->plugLights[i].getBrightness();
+				}
 			}
 		}
+		internal->plugLight->setBrightnesses(values);
 	}
-	internal->plugLight->setBrightnesses(values);
 
 	Widget::step();
 }
diff --git a/src/app/MenuBar.cpp b/src/app/MenuBar.cpp
index 8f9ade8c..93a993fb 100644
--- a/src/app/MenuBar.cpp
+++ b/src/app/MenuBar.cpp
@@ -444,6 +444,8 @@ struct ViewButton : MenuButton {
 
 		menu->addChild(createBoolPtrMenuItem("Lock cursor while dragging params", "", &settings::allowCursorLock));
 
+		menu->addChild(createBoolPtrMenuItem("Disable port lights", "", &settings::disablePortLights));
+
 		static const std::vector<std::string> knobModeLabels = {
 			"Linear",
 			"Scaled linear",
@@ -491,6 +493,16 @@ struct SampleRateItem : ui::MenuItem {
 			[=]() {settings::sampleRate = 0;}
 		));
 
+		// Two special lower sample rates
+		menu->addChild(createCheckMenuItem("16 kHz", "",
+			[=]() {return settings::sampleRate == 16000.f;},
+			[=]() {settings::sampleRate = 16000.f;}
+		));
+		menu->addChild(createCheckMenuItem("32 kHz", "",
+			[=]() {return settings::sampleRate == 32000.f;},
+			[=]() {settings::sampleRate = 32000.f;}
+		));
+
 		// Power-of-2 oversample times 44.1kHz or 48kHz
 		for (int i = -2; i <= 4; i++) {
 			for (int j = 0; j < 2; j++) {
diff --git a/src/engine/Module.cpp b/src/engine/Module.cpp
index 1014c225..1c7bbf73 100644
--- a/src/engine/Module.cpp
+++ b/src/engine/Module.cpp
@@ -373,13 +373,15 @@ void Module::doProcess(const ProcessArgs& args) {
 	}
 
 	// Iterate ports to step plug lights
-	if (args.frame % PORT_DIVIDER == 0) {
-		float portTime = args.sampleTime * PORT_DIVIDER;
-		for (Input& input : inputs) {
-			Port_step(&input, portTime);
-		}
-		for (Output& output : outputs) {
-			Port_step(&output, portTime);
+	if(!settings::disablePortLights) {
+		if (args.frame % PORT_DIVIDER == 0) {
+			float portTime = args.sampleTime * PORT_DIVIDER;
+			for (Input& input : inputs) {
+				Port_step(&input, portTime);
+			}
+			for (Output& output : outputs) {
+				Port_step(&output, portTime);
+			}
 		}
 	}
 }
diff --git a/src/settings.cpp b/src/settings.cpp
index 621a395a..46979382 100644
--- a/src/settings.cpp
+++ b/src/settings.cpp
@@ -31,12 +31,13 @@ float cableTension = 1.0;
 float rackBrightness = 1.0;
 float haloBrightness = 0.25;
 bool allowCursorLock = true;
+bool disablePortLights = false;
 KnobMode knobMode = KNOB_MODE_LINEAR;
 bool knobScroll = false;
 float knobLinearSensitivity = 0.001f;
 float knobScrollSensitivity = 0.001f;
-float sampleRate = 0;
-int threadCount = 1;
+float sampleRate = 22050;
+int threadCount = 2;
 bool tooltips = true;
 bool cpuMeter = false;
 bool lockModules = false;
@@ -45,7 +46,7 @@ bool squeezeModules = true;
 	// Most Mac GPUs can't handle rendering the screen every frame, so use 30 Hz by default.
 	float frameRateLimit = 30.f;
 #else
-	float frameRateLimit = 60.f;
+	float frameRateLimit = 5.f;
 #endif
 float autosaveInterval = 15.0;
 bool skipLoadOnLaunch = false;
@@ -138,6 +139,8 @@ json_t* toJson() {
 
 	json_object_set_new(rootJ, "allowCursorLock", json_boolean(allowCursorLock));
 
+	json_object_set_new(rootJ, "disablePortLights", json_boolean(disablePortLights));
+
 	json_object_set_new(rootJ, "knobMode", json_integer((int) knobMode));
 
 	json_object_set_new(rootJ, "knobScroll", json_boolean(knobScroll));
@@ -307,6 +310,10 @@ void fromJson(json_t* rootJ) {
 	if (allowCursorLockJ)
 		allowCursorLock = json_boolean_value(allowCursorLockJ);
 
+	json_t *disablePortLightsJ = json_object_get(rootJ, "disablePortLights");
+	if (disablePortLightsJ)
+		disablePortLights = json_boolean_value(disablePortLightsJ);
+
 	json_t* knobModeJ = json_object_get(rootJ, "knobMode");
 	if (knobModeJ)
 		knobMode = (KnobMode) json_integer_value(knobModeJ);
